casper/src/test/scala/coop/rchain/casper/engine/LfsStateRequesterStateSpec.scala

```diff
@@ -0,0 +1,113 @@
+package coop.rchain.casper.engine
+
+import coop.rchain.casper.engine.LfsTupleSpaceRequester.ST
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+import org.scalatest.{FlatSpec, Matchers}
+
+class LfsStateRequesterStateSpec extends FlatSpec with Matchers with GeneratorDrivenPropertyChecks {
+
+  "getNext" should "return empty list when called again" in {
+    val st = ST(Seq(10))
```

Comment:
The file here is actually testing the object `ST` which can be not related with LFS, I think. 
Maybe we can move this `ST` in shared folder.MAybe that can be reused in some other cases.

Reply:
You are right. It was completely the same as in block requester but more logic is added for blocks. But for now better to create multiple variants of state object and see what stands out to be shared.
Maybe state can be a trait and different implementations can give us different features.